### ISSUE TITLE
Hide used search parameter

### ISSUE THREAD

**Issue by Seiifer** [dnauck/angular-advanced-searchbox#8]
Hello, and thank you for your work.

I know it is on your todo list from what I saw in the code but I tried to do it myself and can't figure out how to do it right.

First, in my controller I changed the `$scope.availableSearchParams` array by adding a boolean like this : 
```javascript
$scope.availableSearchParams = [
                {key: "name", name: "Name", placeholder: "Name...", isUsed: false},
                {key: "type", name: "Type", placeholder: "Type...", isUsed: false},
                {key: "year", name: "Year", placeholder: "Year...", isUsed: false},
                {key: "area", name: "Area", placeholder: "Sector...", isUsed: false},
                {key: "status", name: "Status", placeholder: "Status...", isUsed: false}
            ];
```

Then I'm trying to edit you `$scope.addSearchParam(...)` function in order to set `isUsed` to true when I push a new search parameter into `$scope.searchParams` and to edit your `$scope.removeSearchParam(...)` function to set it back to false but I can't manage to access `$scope.availableSearchParams`.

But now I'm thinking, even if I could, in order to hide a parameter, we should remove it from `$scope.availableSearchParams`, right ? Unless we can use `isUsed` for each parameter with a `ng-show` in the html ?

In the end, I'm a bit lost, am I getting it all wrong and trying to do it with a bad logic ?
Are you still working on this aspect of the project ?

**Comment by dnauck** [comment]
Hi,

it's on my todo to forbid multiple usage of search parameters.

**Comment by dnauck** [comment]
Related to #6

**Comment by Seiifer** [comment]
Maybe it can be done by simply adding  a `ng-if` in the template here (supposing we have a isUsed status for each parameter) ? :
```html
 <span class="search-parameter" ng-repeat="param in parameters | limitTo:8" ng-mousedown="addSearchParam(param)">{{param.name}}</span>
```

**Comment by dnauck** [comment]
Fixed in 35f051e
